### ISSUE TITLE
Build torch From Source is failed on Mac, maybe caused by protobuf

### ISSUE THREAD

**Issue by mikesongming** [pytorch/pytorch#121215]
### 🐛 Describe the bug

I've tried to build pytorch (v2.2.1) from source to study the source code. However, I was constantly frustrated at the end of  compiling `torch/csrc`, with errors hinting something wrong with onnx/proto. I have found similar error in issue #25580 , but don't know how to fix it.

My logging file uploaded as: 
[install.log](https://github.com/pytorch/pytorch/files/14493308/install.log)

### Versions

Collecting environment information...
PyTorch version: N/A
Is debug build: N/A
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: N/A

OS: macOS 12.7.2 (x86_64)
GCC version: Could not collect
Clang version: 14.0.0 (clang-1400.0.29.202)
CMake version: version 3.22.0
Libc version: N/A

Python version: 3.10.0 (default, Mar  3 2022, 03:57:21) [Clang 12.0.0 ] (64-bit runtime)
Python platform: macOS-10.16-x86_64-i386-64bit
Is CUDA available: N/A
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: N/A

CPU:
Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz

Versions of relevant libraries:
[pip3] numpy==1.26.4
[pip3] optree==0.10.0
[conda] mkl                       2023.1.0         h8e150cf_43560    defaults
[conda] mkl-include               2023.1.0         hecd8cb5_43560    defaults
[conda] numpy                     1.26.4                   pypi_0    pypi
[conda] optree                    0.10.0                   pypi_0    pypi

**Comment by mikesongming** [comment]
fixed using commit in #27583 , recommending to merge it

**Comment by malfet** [comment]
One should be able to build PyTorch without ONNX or protobuf, which will likely solve your build issues

You likely have protobuf installed via homebrew which conflicts with one PyTorch is bundled with

**Comment by mikesongming** [comment]
Yes, I have a preinstalled protobuf v3.21 via homebrew. Is there any chance to fix this problem in CMakeLists.txt? The bug seems common for Mac users, and it caused me so much time...

BTW, "One should be able to build PyTorch without ONNX or protobuf", you mean setting "USE_SYSTEM_ONNX=ON"?
